### PR TITLE
result

### PR DESCRIPTION
**Pull request by ADIGrimm**

### CODE REVIEW

File: src/main/java/online/bookstore/OnlineBookstoreApplication.java [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+import online.bookstore.service.BookService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.CommandLineRunner;
+import org.springframework.boot.SpringApplication;
+import org.springframework.boot.autoconfigure.SpringBootApplication;
+import org.springframework.boot.autoconfigure.domain.EntityScan;
+import org.springframework.context.annotation.Bean;
+
+@SpringBootApplication
+@EntityScan(basePackages = "online.bookstore.model")
```
Comment (Elena-Bruyako) [inline src/main/java/online/bookstore/OnlineBookstoreApplication.java]: remove

File: src/main/java/online/bookstore/OnlineBookstoreApplication.java [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+            book.setPrice(BigDecimal.valueOf(750));
+            book.setTitle("Brave New World");
+            book.setAuthor("Aldous Huxley");
+            book.setDescription("Brave New World is a dystopian social "
+                    + "science fiction novel written by Aldous Huxley. "
+                    + "It presents a future world where society is "
+                    + "controlled by technology and the government.");
+            book.setCoverImage("https://m.media-amazon.com/images/I/81gTwYAhU7L.jpg");
+            book.setIsbn("9780060850524");
+            bookService.save(book);
```
Comment (Elena-Bruyako) [inline src/main/java/online/bookstore/OnlineBookstoreApplication.java:34]: call also findAll method

File: src/main/java/online/bookstore/model/Book.java [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+
+import jakarta.persistence.Column;
+import jakarta.persistence.Entity;
+import jakarta.persistence.GeneratedValue;
+import jakarta.persistence.GenerationType;
+import jakarta.persistence.Id;
+import java.math.BigDecimal;
+import org.antlr.v4.runtime.misc.NotNull;
+
+@Entity
```
Comment (Elena-Bruyako) [inline src/main/java/online/bookstore/model/Book.java:14]:
```suggestion
@Entity
@Getter
@Setter
@Table(name...
```

File: src/main/java/online/bookstore/model/Book.java [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+    private String author;
+    @NotNull
+    @Column(unique = true)
+    private String isbn;
+    @NotNull
+    private BigDecimal price;
+    private String description;
+    private String coverImage;
+
+    public Long getId() {
```
Comment (Elena-Bruyako) [inline src/main/java/online/bookstore/model/Book.java]: remove gettes and setters, you can use Lombok annotations for them

File: src/main/java/online/bookstore/repository/impl/BookRepositoryImpl.java [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+import online.bookstore.model.Book;
+import online.bookstore.repository.BookRepository;
+import org.hibernate.Session;
+import org.hibernate.SessionFactory;
+import org.hibernate.Transaction;
+import org.hibernate.query.Query;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Repository;
+
+@Repository
```
Comment (Elena-Bruyako) [inline src/main/java/online/bookstore/repository/impl/BookRepositoryImpl.java:15]:
```suggestion
@Repository
@RequiredArgsConstructor
```

File: src/main/java/online/bookstore/repository/impl/BookRepositoryImpl.java [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+public class BookRepositoryImpl implements BookRepository {
+    private final SessionFactory sessionFactory;
+
+    @Autowired
+    public BookRepositoryImpl(SessionFactory sessionFactory) {
+        this.sessionFactory = sessionFactory;
+    }
```
Comment (Elena-Bruyako) [inline src/main/java/online/bookstore/repository/impl/BookRepositoryImpl.java]:
```suggestion
```

File: src/main/java/online/bookstore/repository/impl/BookRepositoryImpl.java [hunk 7]
```diff
@@ -0,0 +1,53 @@
...
+            session = sessionFactory.openSession();
+            transaction = session.beginTransaction();
+            session.persist(book);
+            transaction.commit();
+            return book;
+        } catch (Exception e) {
+            if (transaction != null) {
+                transaction.rollback();
+            }
+            throw new RuntimeException("Can't add a book to db: " + book, e);
```
Comment (Elena-Bruyako) [inline src/main/java/online/bookstore/repository/impl/BookRepositoryImpl.java]:
```suggestion
            throw new DataProcessingException("Can't add a book to db: " + book, e);
```

File: src/main/java/online/bookstore/repository/impl/BookRepositoryImpl.java [hunk 8]
```diff
@@ -0,0 +1,53 @@
...
+        }
+    }
+
+    @Override
+    public List<Book> findAll() {
+        try (Session session = sessionFactory.openSession()) {
+            Query<Book> getAllBooksQuery = session.createQuery("FROM Book", Book.class);
+            return getAllBooksQuery.getResultList();
+        } catch (Exception e) {
+            throw new RuntimeException("Can't get list of all books" + e);
```
Comment (Elena-Bruyako) [inline src/main/java/online/bookstore/repository/impl/BookRepositoryImpl.java]:
```suggestion
            throw new DataProcessingException("Can't get list of all books" + e);
```

File: src/main/java/online/bookstore/service/impl/BookServiceImpl.java [hunk 9]
```diff
@@ -0,0 +1,28 @@
+package online.bookstore.service.impl;
+
+import java.util.List;
+import online.bookstore.model.Book;
+import online.bookstore.repository.BookRepository;
+import online.bookstore.service.BookService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+@Service
```
Comment (Elena-Bruyako) [inline src/main/java/online/bookstore/service/impl/BookServiceImpl.java:10]:
```suggestion
@Service
@RequiredArgsConstructor
```

File: src/main/java/online/bookstore/service/impl/BookServiceImpl.java [hunk 10]
```diff
@@ -0,0 +1,28 @@
...
+public class BookServiceImpl implements BookService {
+    private final BookRepository bookRepository;
+
+    @Autowired
+    public BookServiceImpl(BookRepository bookRepository) {
+        this.bookRepository = bookRepository;
+    }
```
Comment (Elena-Bruyako) [inline src/main/java/online/bookstore/service/impl/BookServiceImpl.java]:
```suggestion
```

**Review by Elena-Bruyako — CHANGES_REQUESTED** [review]
see comments

**Review by sarakhmen — APPROVED** [review]

**Pull request merged**